activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -680,11 +680,37 @@ def add_timestamps_for_alter(table_name, options = {})
             options[:precision] = 6
           end
 
-          [add_column_for_alter(table_name, :created_at, :datetime, options), add_column_for_alter(table_name, :updated_at, :datetime, options)]
+          only_option = options.delete(:only)
+
+          if only_option == :created_at
+            return [add_column_for_alter(table_name, :created_at, :datetime, options)]
+          end
+
+          if only_option == :updated_at
+            return [add_column_for_alter(table_name, :updated_at, :datetime, options)]
+          end
+
+          [
+            add_column_for_alter(table_name, :created_at, :datetime, options),
+            add_column_for_alter(table_name, :updated_at, :datetime, options)
+          ]
```

Comment:
This method would then be:

```ruby
def add_timestamps_for_alter(table_name, *timestamps, **options)
  options[:null] = false if options[:null].nil?

  if !options.key?(:precision) && supports_datetime_with_precision?
    options[:precision] = 6
  end
  
  timestamps = %i[ created_at updated_at ] if timestamps.empty?
  timestamps.map { |timestamp| add_column_for_alter(table_name, timestamp, :datetime, options) }
end
```

Similar technique would be used on the other implementation.